doc/port-add-del-dynamically/dynamic_port_add_del_hld.md

```diff
@@ -0,0 +1,305 @@
+
+**No need to change the code**
+
+#### Sflowmgr:
+Add port: Event from config db - Update the speed to sflow internal db.<br />
+Del port: Delete event from config db - remove the speed from sflow internal db.<br />
+
+**No need to change the code**
+
+#### Teammgrd:
+Listen to events from state db, when entry is added -> add the port to lag<br />
+
+**No need to change the code**
+
+#### Macsecmgr:
+Listen to events on cfg port table – the service will enable or disable macsec if macsec was configured on the port cfg table (using the macsec field)<br />
+
+**No need to change the code**
+
+#### snampagent:
+•   Add/remove port has no special treatment.<br />
+each time the snmpagent needs information from ports (oper_state, mtu, speed..) it reads from APP port table. Will be triggered on mib requests.<br />
+
+**No need to change the code**
+
+#### PMON - Xcvrd:
+Listen to events on cfg port table and update transeiver information <br />
+
+implemented on those PRs:
+https://github.com/Azure/sonic-buildimage/pull/8422 <br />
+https://github.com/Azure/sonic-platform-daemons/pull/212
+
+
+## Buffermgrd:
+
+##### Add port:  
+- If a port is added without a buffer configuration the buffer configuration the SDK will “decide” the default buffer values for this port.
+- The user can add the port on admin state down -> later add the buffer configuration (static or dynamic) -> enable this port.
+For example, in the line-card system case:
+-   When line card is provisioned the line card manager is adding a port to config db, need to add the port with admin state “down”.
+-   Line card manager will add the buffer configuration for this port through a default buffer cfg template.
+-   Line cart manager will enable the port.
+
+•   Pg_profile_lookup file has values that will be used for static buffer configuration.
+for each port speed and cable length we have buffer size value, xon and xoff value <br />
+For example:
+
+
+|speed cable | cable | size  |  xon  |  xoff |  threshold |
+|:----------:|:-----:|:-----:|:-----:|:-----:|:----------:|
+|  10000     | 5m    | 49152 | 19456 | 29696 |      0     |
+|  25000     | 5m    | 49152 | 19456 | 29696 |      0     |
+|  40000     | 5m    | 49152 | 19456 | 29696 |      0     |
+|  50000     | 5m    | 49152 | 19456 | 29696 |      0     |
+
+On the line-card system we will use different types of line cards (maybe with different gearboxes), the values on the pg_profile_lookup will be used for all the types.<br />
+we may need to consider using pg_profile_lookup.ini for each line card type.<br />
+•   When port is added to the config db – the speed and the admin state is saved on internal db<br />
+•   After port was added the user can add buffer configuration to this port (dynamic or static configuration) and only then the buffermgr will set the buffer configuration on App table<br />
+
+•   We have rare situation of race condition in the add port flow:<br />
```

Comment:
when applying buffer cfg into port, we should always check if port is ready first, so this race condition should be handled.  If this check is not done, we should do it.

Reply:
there is no check if port exist, we can check it before trying to add buffer cfg, if it is not exist need to retry.
I will add it

Reply:
added to the document